Client/Frontend/Browser/URIFixup.swift

```diff
@@ -34,4 +39,29 @@ class URIFixup {
 
         return nil
     }
-}
\ No newline at end of file
+
+    static func strToAscii(string : String) -> String {
+        var host : String?
+
+        if string.hasPrefix("http://") {
+            host = string.substringFromIndex(string.startIndex.advancedBy(7))
+        } else if string.hasPrefix("https://") {
+            host = string.substringFromIndex(string.startIndex.advancedBy(8))
+        } else {
+            host = string
+        }
+
+        let t = host!.rangeOfString("/")
+        if (t?.startIndex) != nil {
+            host = host?.substringToIndex((t?.startIndex)!)
+        }
+
+        let s = host!.rangeOfString(":")
+        if (s?.startIndex) != nil {
+            host = host?.substringToIndex((t?.startIndex)!)
+        }
+
+        let urlStr = host?.utf8HostToAscii()
+        return urlStr!
```

Comment:
I think we might be able to remove this function altogether. What if we instead do something like this:

```
    static func punycodedURL(string: String) -> NSURL? {
        let components = NSURLComponents(string: string)
        components?.host = components?.host?.utf8HostToAscii()
        return components?.URL
    }
```

That way, we let `NSURLComponents` handle all the parsing work. We just have to be sure to pass the escaped URL to this function so that the URL parse succeeds.